### PR TITLE
examples and returns

### PR DESCRIPTION
**Pull request by scottnath**
Added code for [@examples](http://usejsdoc.org/tags-example.html) and made a possibility for multiple returns.

### CODE REVIEW

File: lib/parser.js [hunk 1]
```diff
@@ -98,9 +100,18 @@ function _mapSymbol(symbol){
     symbol.type = types[0].types.join(' | ');
   }
 
-  if(returns.length === 1){
-    symbol["return"] = returns[0].types.join(' ');
-    symbol["returnDescription"] = returns[0].description.replace(/(<([^>]+)>)/ig, '');
```
Comment (FGRibreau) [inline lib/parser.js]: why did you remove `returnDescription`?

Comment (scottnath) [inline lib/parser.js]: Because the possibility exists for multiple returns. For instance, see [this file](https://github.com/pattern-library/pattern-importer/blob/master/lib/import-single-pattern.js) where the returned object is broken down into its various parts. With a single return/returnDescription there is no way to do that, each subsequent return would overwrite the last. Bad coding from me in the first place.

Also, for my second go at this file, I wanted to clean up my changes, making them more uniform - thus the change from isReturn/isRequires to hasReturn/hasRequires as well. And now the return section matches the way examples & requires work.

**Comment by scottnath** [comment]
@FGRibreau I also added in file/fileoverview/overview tags. These are oddly not natively supported in dox, I've [submitted an issue](https://github.com/tj/dox/issues/169) there to see what it would take to get them added in.

**Comment by scottnath** [comment]
dox fixed the bug and upgraded to 0.8.0
https://github.com/tj/dox/issues/169

File: lib/parser.js (hunk 1 above)
Comment (FGRibreau) [inline lib/parser.js]: Indeed! Could you please update the template as well (removing returnDescription for instance)

Comment (scottnath) [inline lib/parser.js]: Hey @FGRibreau *returnDescription* never made it into the template. I only edited parser.js

this is my @return section in my local version of the jade template:

## @return
```
if symbol.returns
  table.table.table-bordered.table-striped
    thead
      tr
        th(style="width:20%") @return
        th Description
    tbody
      each tag in symbol.tags
        if tag.type == 'return'
          tr
            td= tag.types.toString()
            td= tag.description.replace(/(<([^>]+)>)/ig, '')
```

Comment (scottnath) [inline lib/parser.js]: current jade template:
https://github.com/scottnath/pattern-library-utilities/blob/master/lib/templates/doxx.template.jade

My plan is to write something which can parse the @requires section to link out to NPM or to local files. Before I do...ever seen anything like that?

Comment (FGRibreau) [inline lib/parser.js]: Nice ! Nope, I don't know any but for the link use `http://ghub.io/{packageName}` :)

**Comment by FGRibreau** [comment]
@scottnath it's OK for me, however there seems to be some conflict, can you rebase?

**Comment by scottnath** [comment]
@FGRibreau did a pull from FGRibreau/doxx and merged in.
